08-regression2.Rmd

```diff
@@ -195,7 +199,7 @@ lm_test_results
 ```
 
 Our final model's test error as assessed by RMSPE
```

Comment:
This is the third time we are calculating the test error for a model for this test data set... Again, I think we need to have a note that this is just for teaching purposes (In practice, we should be doing cross validation and comparing cross validated errors...)

Reply:
it is OK to compare test error as long as you aren't using it for model training purposes. Like basically at the end of a paper you should compare multiple methods (past work) and your own new shiny classifier using the test set. That being said, the more methods you compare, the less reliable the comparison is. But just two or three methods is probably fine.

I did add text in the knn regression section about if you want to choose between single + multi knn *as part of your model design process*, you should only be using the training data to do that. I'll do the same thing in linreg.